src/lib.rs

```diff
@@ -40,7 +39,7 @@ impl<T: Copy> Trailer<T> {
 impl<T> Trailer<T> {
     unsafe fn allocate(capacity: usize) -> Trailer<T> {
         let size = mem::size_of::<T>() + capacity;
-        let align = mem::align_of::<u8>();
+        let align = mem::align_of::<T>();
```

Comment:
This is required to make sure the pointer is aligned to `T`, otherwise you can't write `T` into it.